src/observable/ConnectableObservable.ts

```diff
@@ -25,7 +25,16 @@ export class ConnectableObservable<T> extends Observable<T> {
     return (this.subject = this.subjectFactory());
   }
 
-  connect(): Subscription<T> {
+  connect(onSubscribe?: (subscription: Subscription<T>) => void): Subscription<T> {
+    if (onSubscribe) {
+      this._callbackConnect(onSubscribe);
+      return null;
```

Comment:
@staltz I'm not sure, just thinking about whether somebody can call `connect` like this publicly, and whether `onSubscribe` will always synchronously invoked. If there's any (potential) asynchrony at all, we should have a `Subscription` to track it.